### DISCUSSION TITLE
Convert C++ code to C# for quick integration into a software in development

### DISCUSSION THREAD

**Discussion by Elvis254** [dotnet/roslyn#74967] [Q&A] [answered]
Hi @333fred,
Am writing a software used to manage elections in Kenya using C#. The client is working on a tight schedule and wants me to complete the work before Monday. Is there a way I can be able to convert C++ code to C# in order to quickly add biometric capabilities to the software? Here is the link of the code I want converted to C#: [https://learn.microsoft.com/en-us/windows/win32/secbiomet/biometric-service-api-portal](https://learn.microsoft.com/en-us/windows/win32/secbiomet/biometric-service-api-portal)
Elvis.

**Comment by 333fred** [comment, marked as answer]
I don't know of any such converters, and this really isn't related to Roslyn. While this is a discussion forum, it's a discussion forum for the Roslyn C# and VB compilers, not a general help forum. If you need more general help with C#, you can try going to a community space like the C# community discord server, rather than a specific project repository.

**Reply by Elvis254** [reply]
What I wanted is a documentation written showing C# code samples. I have found the C# community discord server [https://discord.gg/csharp](https://discord.gg/csharp) and joined it. Back in the years we used to convert VB6 code to VB.NET code, I think you didn't understand what I wanted but if the C++ code samples are given side to side with C# it will have helped me.

**Comment by paul1956** [comment]
There are converters free and paid to convert from many languages to others,  including to and from C++, C#, Java VB… most require some manual editing but you will get close. Just do a web search. Free versions usually do 1 file up to 100 lines. Paid ones have no limits and co do entire solutions. Sent from my iPhoneI apologize for any typos Siri might have made.(503) 803-6077On Sep 1, 2024, at 12:49 AM, Masaba Elvis ***@***.***> wrote:﻿
What I wanted is a documentation written showing C# code samples. Please, send me an example of this C# community discord server. Back in the years we used to convert VB6 code to VB.NET code, I think you didn't understand what I wanted but if the C++ code samples are given side to side with C# it will have helped me.

**Reply by Elvis254** [reply]
Hi @paul1956,
I found a paid one and used it to convert one of the code samples here is the link: [https://www.codeconvert.ai/](https://www.codeconvert.ai/) it accurately converted the code from C++ to C#.
Elvis.
